server/src/main/java/io/crate/expression/scalar/regex/RegexMatcher.java

```diff
@@ -22,70 +22,37 @@
 package io.crate.expression.scalar.regex;
 
 import javax.annotation.Nullable;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-public class RegexMatcher {
+public abstract class RegexMatcher {
```

Comment:
Minor: `RegexpMatcher` would be more consistent with the rest of the crate naming.